lib/compress/huf_compress.c

```diff
@@ -44,9 +44,50 @@
 /* **************************************************************
 *  Utils
 ****************************************************************/
-unsigned HUF_optimalTableLog(unsigned maxTableLog, size_t srcSize, unsigned maxSymbolValue)
+size_t HUF_getCTableSize(const HUF_CElt* CTable, unsigned maxSymbolValue, unsigned huffLog);
+
+#define IMPROVEMENT_THRESHOLD 5 /* estimation can be off by 4 bytes, only shift if we're sure we're winning */
+#define ESTIMATE_SIZE(c, result) CHECK_F(HUF_buildCTable(ct, count, maxSymbolValue, c)); \
+    result = HUF_estimateCompressedSize(ct, count, maxSymbolValue) + HUF_getCTableSize(ct, maxSymbolValue, c);
```

Comment:
minor : 
consider using the _`wksp` variants before merging this PR.
It looks to me that the "normal" variant which are invoked here use some stack space.
For general use cases, it's not significant.
But for kernel scenarios, anything > 100 bytes is significant, and should use heap memory instead.